Makefile-libostree-defines.am

```diff
@@ -48,6 +47,12 @@ libostree_public_headers = \
 	src/libostree/ostree-kernel-args.h \
 	$(NULL)
 
+if USE_GPGME
+libostree_public_headers += \
+	src/libostree/ostree-gpg-verify-result.h \
```

Comment:
I think we probably need to keep installing this, and write a `ostree-gpg-verify-result-dummy.c` or so instead that's compiled if GPG is disabled?

Basically aim to keep the same public API, just make GPG operations an error.  We should also make it easy for apps to detect if libostree doesn't support GPG, maybe return `G_IO_ERROR_NOT_IMPLEMENTED` or so.

Reply:
Ok. Will add that.

Reply:
Should we support the "classic" CLI interface as well? i.e. command 'gpg_sign' and gpg-related options for 'commit', 'pull' etc.?

Reply:
I'm uncertain.  Offhand...it seems nice if we can avoid having a different API/ABI depending on whether gpg is on or off.  But, I don't think maintaining the same ABI is a hard requirement either.

In practice, I expect most general purpose distributions to continue to enable GPG, and people who want to turn it off will do so globally.  However, it does highlight that dependent software like flatpak will also need to learn how to deal with this too.

Probably, we need to *generate* data in `ostree.h` with our build-time feature flags.  We can't use `#ifdef HAVE_GPG` there, that's what's breaking the CI. 
What we should do instead at build time generate `ostree-build-options.h` or so, that does `#define OSTREE_BUILDOPT_GPG` if it's enabled, and then install that header.

That way inside our own public headers we can use it, and so can dependent software.

Reply:
The alternative is to maintain the same public API, and just return errors.

Reply:
Changes:
- `src/libostree/ostree-gpg-verify-result.h` is installed
- added `ostree-gpg-verify-result-dummy.c` with dummy public functions
- some unneeded "ifdef"s removed since I do not remove public API anymore
- rename define from "HAVE_GPGME" to "OSTREE_DISABLE_GPGME"

After checking of CLI utilities I decide to keep the removing of util and gpg-related options. At least we already have the same behavior for "curl", so...

believe the generation of `ostree.h` is bit out of scope of current PR.
However I think it would be enough to rename already generated `config.h` into `ostree-build-options.h` and include it to installables.

Reply:
@cgwalters any thought about updated version?

Reply:
OK related to the detection, I did https://github.com/ostreedev/ostree/pull/1890

Reply:
hm.... does it means I need to return back logic "#ifdef HAVE_GPGME" and hope downstream will check if feature is enabled?

Reply:
The main thing we *needed* to fix is having `#ifdef HAVE_GPGME` in our public headers.  It just can't work.  And I believe your current PR has addressed that.

The next problem is though; apps like flatpak probably want to detect whether or not libostree supports gpg.   The PR I did isn't quite ergonomic for C code to do build-time checking; can't do `strstr()` in the C preprocessor.  I think we'd have to explode the feature flags into `#defines`.  But it was easy to do #1890 and it does allow runtime dispatch at least.

Reply:
> The main thing we _needed_ to fix is having `#ifdef HAVE_GPGME` in our public headers. It just can't work. And I believe your current PR has addressed that.

but it is already fixed. In last available PR I did not touch public headers.
Sorry for force pushing to the source branch -- but otherwise changes would be even worse to read since I have to rework `ifdef` logic to `ifndef`

> The next problem is though; apps like flatpak probably want to detect whether or not libostree supports gpg. 

yeah, that's why I did that reversion to `#ifndef OSTREE_DISABLE_GPGME`  -- now all apps just build and dummy GPG stuff will throw an error in runtime if there is no GPGME support but it is requested by app.
So the fixed version shouldn't break public API.